dbt/adapters/snowflake/connections.py

```diff
@@ -209,19 +210,23 @@ def _get_access_token(self) -> str:
         return result_json["access_token"]
 
     def _get_private_key(self):
-        """Get Snowflake private key by path or None."""
-        if not self.private_key_path:
-            return None
-
+        """Get Snowflake private key by path, from a Base64 encoded DER bytestring or None."""
         if self.private_key_passphrase:
             encoded_passphrase = self.private_key_passphrase.encode()
         else:
             encoded_passphrase = None
 
```

Comment:
Would prefer to be more explicit here and raise an exception if both are supplied: 
```suggestion
if self.private_key and self.private_key_path:
   raise DbtProfileError("Cannot specify both `private_key`  and `private_key_path`")
elif  self.private_key:
```